tests/functional/test_new_resolver.py

```diff
@@ -414,6 +414,29 @@ def test_new_resolver_requires_python_error(script: PipTestEnvironment) -> None:
     assert message in result.stderr, str(result)
 
 
+def test_new_resolver_requires_python_ok_with_python_version_flag(
+    script: PipTestEnvironment,
+) -> None:
+    create_basic_wheel_for_package(
+        script,
+        "base",
+        "0.1.0",
+        requires_python="<3",
+    )
+    result = script.pip(
```

Comment:
This currently fails due to https://github.com/pypa/pip/issues/12215, once https://github.com/pypa/pip/pull/12217 is in, I'll merge `main` back into this branch.